### ISSUE TITLE
(probably) erroneous warning

### ISSUE THREAD

**Issue by robinmoussu** [Xudong-Huang/generator-rs#22]
When some of the values generated by a generator created by `Gn` are not consumed, a warning appears in the logs.

I assumed this warning is [erroneous](https://docs.rs/generator/0.6.21/src/generator/gen_impl.rs.html#489), and should at most be a `trace!()` if any.

```rust
use generator::*;

// This generator seems totally fine
fn generate_values(start: u8) -> impl Iterator<Item = u8> {
    Gn::new_scoped(move |mut s| { // The warning will only appears if it's a moved closure
        for i in start..20 {
            s.yield_(i);
        }
        done!()
    })
}

fn main() {
    env_logger::init();
    let _: Vec<_> = dbg!(
        generate_values(3).
        .take(3) // this will abort the generator early
        .collect());
}
```

```toml
# Cargo.toml
[dependencies]
log = "0.4"
env_logger = "0.7.1"
```

```
$ RUST_LOG=trace cargo run
[2020-07-24T15:45:33Z WARN  generator::gen_impl] generator is not done while drop
[src/main.rs:26] foo(3).take(3).collect() = [
    3,
    4,
    5,
]
```

**Comment by Xudong-Huang** [comment]
Make sense. I will update it in next release.

**Comment by robinmoussu** [comment]
@Xudong-Huang I think you forgot to modify this in 0.6.22.

**Comment by Xudong-Huang** [comment]
released in 0.6.23
